test/function/target/stream.cpp

```diff
@@ -24,6 +24,7 @@
 #include <gtest/gtest.h>
 #include "test/helper/assert.hpp"
 #include "test/helper/rfc5424.hpp"
+#include "private/target/stream.h"
```

Comment:
@abhakta-47  This line is likely the cause for your compilation issues: private headers are not supported outside of the library compilation itself. Your tests need to work without acceess to the internals of the structures, as a result ("black box" tests).